### DISCUSSION TITLE
Sentry Support

### DISCUSSION THREAD

**Discussion by vmesel** [prest/prest#814] [API Server]
As I need to add some monitoring to the server, I would like pREST to add support to Sentry so I can get application errors on my Sentry dashboard.

**Comment by arxdsilva** [comment]
Good idea, it could be done either by the new middleware plugin system or doing a new custom configuration.

I personally would do a new configuration, just make sure that it is optional and can work with multiple providers, but only one at a time.

**Reply by avelino** [reply]
the *question we need to ask if it should be a plugin or in the core* **is if everyone would use sentry**, if it is in the core to support others than sentry we are ~unnecessarily inflating the core~

I like the idea of it being **prestd's official plugin**, where we develop and maintain

> It is a great plugin for us to evolve the plugin support - it is very initial and we need "real cases" to improve it

**Reply by vmesel** [reply]
So I've attempted to build a pREST plugin for Sentry. Using the base hello.go code, I just renamed it to sentry.go and changed the function name inside the code.

There are some huge problems while compiling it and developing it:

First, and most importantly the negroni/v3 version, even using the same go.mod and go.sum as used in pREST, gives us an error: `plugin.Open("/app/lib/middlewares/sentry"): plugin was built with a different version of package github.com/urfave/negroni/v3`

Second, and not least, our current source code disables the error output to an HTTP request, so whenever we are sending an error, [it only appears on pREST CLI, not on the browser (not even when Debug mode is on)](https://github.com/prest/prest/blob/main/controllers/sql.go#L36-L37), so there is no other way to get the error to the plugin.

**Reply by arxdsilva** [reply]
@vmesel can you make a PR with your plugin attempt?

**Reply by avelino** [reply]
@vmesel share sentry plugin code here https://github.com/prest/sentry-prestd-plugin

**Reply by vmesel** [reply]
There you are @avelino @arxdsilva https://github.com/prest/sentry-prestd-plugin/tree/base-code
